XiEditor/AppDelegate.swift

```diff
@@ -144,25 +144,23 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
     // The default name for XiEditor's error logs
     let defaultCoreLogName = "xi_tmp.log"
     
-    lazy var errorLogDirectory: URL = {
+    lazy var errorLogDirectory: URL? = {
         let logDirectory = FileManager.default.urls(
-        for: .libraryDirectory,
-        in: .userDomainMask)
-        .first!
-        .appendingPathComponent("Logs")
-        .appendingPathComponent("XiEditor")
+            for: .libraryDirectory,
+            in: .userDomainMask)
+            .first!
+            .appendingPathComponent("Logs")
+            .appendingPathComponent("XiEditor")
         
         // create XiEditor log folder on first run
-        if !FileManager.default.fileExists(atPath: logDirectory.path) {
-            do {
-                try FileManager.default.createDirectory(at: logDirectory,
-                                                        withIntermediateDirectories: true,
-                                                        attributes: nil)
-            } catch let err as NSError {
-                print("failed to create error log directory. \(err)")
-            }
+        do {
+            try FileManager.default.createDirectory(at: logDirectory,
+                                                    withIntermediateDirectories: true,
+                                                    attributes: nil)
+        } catch let err as NSError {
+            print("failed to create error log directory. \(err)")
+            return nil
```

Comment:
we don't need to handle the error if we're okay being nil; so this can just be something like `return try? FileManager...`